experimenter/experimenter/experiments/api/v5/serializers.py

```diff
@@ -1507,18 +1507,20 @@ def _validate_rollout_version_support(self, data):
         return data
 
     def _validate_bucket_duplicates(self, data):
-        if not self.instance or not self.instance.is_rollout:
-            return data
         is_rollout = self.instance.is_rollout
+        if not self.instance or not is_rollout:
+            return data
+
         count = NimbusExperiment.objects.filter(
             status=NimbusExperiment.Status.LIVE,
             channel=self.instance.channel,
             application=self.instance.application,
             targeting_config_slug=self.instance.targeting_config_slug,
+            feature_configs__in=self.instance.feature_configs.all(),
             is_rollout=is_rollout,
         ).count()
 
-        if is_rollout and count > 0:
+        if count > 0:
```

Comment:
Actually another neat trick you could do, but this is super minor, is

```py
# create a queryset object, but don't evaluate it (ie it's not actually executed in the db)
matching_rollouts = NimbusExperiment.objects.filter(....)

# now the query is executed, and since we only care if any exist we don't even need to count, just check existence which is an even lighter db call
if matching_rollouts.exists():
    return data
```

Reply:
Oh shoot I didn't see this before I merged. Thanks for the tips!! I might open a new PR for that once I'm done this other work I have in progress 🙌